sys/include/xtimer.h

```diff
@@ -0,0 +1,432 @@
+ *
+ * @return 1 on success
+ * @return 0 when timer was not active
+ */
+int xtimer_remove(xtimer_t *timer);
+
+/**
+ * @brief receive a message blocking but with timeout
+ *
+ * @param[out]   m           pointer to a msg_t which will be filled in case of no timeout
+ * @param[in]    us          timeout in microseconds relative
+ *
+ * @return       < 0 on error, other value otherwise
+ */
+int xtimer_msg_receive_timeout(msg_t *m, uint32_t us);
+
+/**
+ * @brief receive a message blocking but with timeout, 64bit version
+ *
+ * @param[out]   m           pointer to a msg_t which will be filled in case of no timeout
+ * @param[in]    us          timeout in microseconds relative
+ *
+ * @return       < 0 on error, other value otherwise
+ */
+int xtimer_msg_receive_timeout64(msg_t *m, uint64_t us);
+
+/**
+ * @brief xtimer backoff value
+ *
+ * All timers that are less than XTIMER_BACKOFF microseconds in the future will
+ * just spin.
+ *
+ * This is supposed to be defined per-device in e.g., periph_conf.h.
+ */
+#ifndef XTIMER_BACKOFF
+#define XTIMER_BACKOFF 30
+#endif
+
+/**
+ * @brief xtimer overhead value
+ *
+ * This value specifies the time a timer will be late if uncorrected.
+ *
+ * E.g., with XTIMER_OVERHEAD == 0
+ * now=xtimer_now();
+ * xtimer_set(&timer, X);
+ * overhead=X-now
+ *
+ * xtimer automatically substracts XTIMER_OVERHEAD from a timer's target time.
+ *
+ * This is supposed to be defined per-device in e.g., periph_conf.h.
+ */
+#ifndef XTIMER_OVERHEAD
+#define XTIMER_OVERHEAD 20
+#endif
+
+#ifndef XTIMER_ISR_BACKOFF
+/**
+ * @brief   xtimer isr backoff time
+ *
+ * When scheduling the next isr, if it is less than the backoff time
+ * in the future, just spin.
+ */
+#define XTIMER_ISR_BACKOFF 20
+#endif
+
+/**
+ * @brief set xtimer default timer configuration
+ * @{
+ */
+#ifndef XTIMER
+#define XTIMER (0)
+#define XTIMER_CHAN (0)
+
+#if TIMER_0_MAX_VALUE == 0xffffff
+#define XTIMER_MASK 0xff000000
+#elif TIMER_0_MAX_VALUE == 0xffff
+#define XTIMER_MASK 0xffff0000
+#endif
+
+#endif
+/**
+ * @}
+ */
+
+#ifndef XTIMER_MASK
+/**
+ * @brief xtimer timer mask
+ *
+ * This value specifies the mask relative to 0xffffffff that the used timer
+ * counts to, e.g., 0xffffffff & ~TIMER_MAXVALUE.
+ *
+ * For a 16bit timer, the mask would be 0xFFFF0000, for a 24bit timer, the mask
+ * would be 0xFF000000. Don't set this for 32bit timers.
+ *
+ * This is supposed to be defined per-device in e.g., periph_conf.h.
+ */
+#define XTIMER_MASK 0
+#endif
+
+#ifndef XTIMER_USLEEP_UNTIL_OVERHEAD
+/**
+ * @brief xtimer_usleep_until overhead value
+ *
+ * This value specifies the time a xtimer_usleep_until will be late
+ * if uncorrected.
+ *
+ * This is supposed to be defined per-device in e.g., periph_conf.h.
+ */
+#define XTIMER_USLEEP_UNTIL_OVERHEAD 10
+#endif
+
+#if XTIMER_MASK
+extern volatile uint32_t _high_cnt;
+#endif
+
+/**
+ * @brief IPC message type for xtimer msg callback
+ */
+#define MSG_XTIMER 12345
```

Comment:
At least reusing the vtimer msg type will not cause any regressions for applications which use custom message types after switching from vtimer to xtimer, and it is unlikely that the same process will be expecting both xtimer and vtimer messages at the same time (or it was written by a very confused developer).